replica/replica.go

```diff
@@ -243,6 +248,32 @@ func IsHeadDisk(diskName string) bool {
 	return false
 }
 
+func (r *Replica) setUUID() string {
+	r.uuidLock.Lock()
+	defer r.uuidLock.Unlock()
+	if r.UUID == "" {
+		const uuidFileName = "uuid"
+		buf := make([]byte, 36)
+		if _, err := os.Stat(r.diskPath(uuidFileName)); err != nil {
+			uuidFile, _ := os.Create(r.diskPath(uuidFileName))
+			r.UUID = util.UUID()
+			copy(buf[:], r.UUID)
+			_, err = uuidFile.Write(buf)
+			logrus.Infof("wrote new uuid in file with error", err)
```

Comment:
it will be good if we catch error and return it to the callee ?
Replica's should not continue in case of error. It should be errored out.

Reply:
Will do